[BUG] Unable to get past splash screen on Android

**Describe the bug**
Users are reporting they are unable to get past the splash screen on Android.


**Additional context**
Here are some stack traces from Play Console, but I don't know they are related, they just coincide with the reports we've been getting. This is affecting many users.

Data comes from Android devices whose users have opted in to automatically share their usage and diagnostics data.


This is the most consistent stack trace from Play Store users.
```
java.lang.RuntimeException: 
  at android.app.ActivityThread.handleCreateService (ActivityThread.java:4412)
  at android.app.ActivityThread.access$1800 (ActivityThread.java:274)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2118)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:237)
  at android.app.ActivityThread.main (ActivityThread.java:8167)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:496)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1100)
Caused by: java.lang.IllegalStateException: 
  at io.flutter.embedding.engine.loader.FlutterLoader.ensureInitializationComplete (FlutterLoader.java:174)
  at io.flutter.view.FlutterMain.ensureInitializationComplete (FlutterMain.java:78)
  at io.flutter.plugins.firebasemessaging.FlutterFirebaseMessagingService.onCreate (FlutterFirebaseMessagingService.java:71)
  at android.app.ActivityThread.handleCreateService (ActivityThread.java:4400)
  at android.app.ActivityThread.access$1800 (ActivityThread.java:274)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2118)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:237)
  at android.app.ActivityThread.main (ActivityThread.java:8167)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:496)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1100)
```
This is the second most consistent reported stack trace from Play Store users.
```
java.lang.RuntimeException: 
  at android.app.ActivityThread.handleCreateService (ActivityThread.java:3500)
  at android.app.ActivityThread.-wrap4 (Unknown Source)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1756)
  at android.os.Handler.dispatchMessage (Handler.java:105)
  at android.os.Looper.loop (Looper.java:164)
  at android.app.ActivityThread.main (ActivityThread.java:6695)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:240)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:772)
Caused by: java.lang.IllegalStateException: 
  at io.flutter.embedding.engine.loader.FlutterLoader.ensureInitializationComplete (FlutterLoader.java:174)
  at io.flutter.view.FlutterMain.ensureInitializationComplete (FlutterMain.java:78)
  at io.flutter.plugins.firebasemessaging.FlutterFirebaseMessagingService.onCreate (FlutterFirebaseMessagingService.java:71)
  at android.app.ActivityThread.handleCreateService (ActivityThread.java:3490)
  at android.app.ActivityThread.-wrap4 (Unknown Source)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1756)
  at android.os.Handler.dispatchMessage (Handler.java:105)
  at android.os.Looper.loop (Looper.java:164)
  at android.app.ActivityThread.main (ActivityThread.java:6695)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:240)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:772)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
+### Fixed
+- Crashes with FCM notifications in Android [#213](https://github.com/rokwire/illinois-app/issues/213)
 
 ## [2.3.30] - 2021-1-18
 ### Changed

--- a/android/app/src/main/java/edu/illinois/rokwire/App.java
+++ b/android/app/src/main/java/edu/illinois/rokwire/App.java
@@ -31,6 +31,7 @@ import androidx.lifecycle.Lifecycle;
 import androidx.lifecycle.LifecycleObserver;
 import androidx.lifecycle.OnLifecycleEvent;
 import androidx.lifecycle.ProcessLifecycleOwner;
+import io.flutter.view.FlutterMain;
 
 public class App extends Application implements LifecycleObserver {
 
@@ -41,6 +42,7 @@ public class App extends Application implements LifecycleObserver {
     @Override
     public void onCreate() {
         super.onCreate();
+        FlutterMain.startInitialization(this);
 
         ProcessLifecycleOwner.get().getLifecycle().addObserver(this);
         createNotificationChannel(this);

--- a/android/app/src/main/java/edu/illinois/rokwire/MainActivity.java
+++ b/android/app/src/main/java/edu/illinois/rokwire/MainActivity.java
@@ -66,10 +66,8 @@ import io.flutter.embedding.engine.FlutterEngine;
 import io.flutter.plugin.common.MethodCall;
 import io.flutter.plugin.common.MethodChannel;
 import io.flutter.plugin.common.PluginRegistry;
-import io.flutter.plugin.common.StandardMessageCodec;
 import io.flutter.plugins.firebasemessaging.FirebaseMessagingPlugin;
 import io.flutter.plugins.firebasemessaging.FlutterFirebaseMessagingService;
-import io.flutter.view.FlutterMain;
 
 public class MainActivity extends FlutterActivity implements MethodChannel.MethodCallHandler, PluginRegistry.PluginRegistrantCallback {
 
@@ -101,7 +99,6 @@ public class MainActivity extends FlutterActivity implements MethodChannel.Metho
 
         // TODO: Check do we need the next two lines at all?
         FlutterFirebaseMessagingService.setPluginRegistrant(this);
-        FlutterMain.startInitialization(this);
     }
 
     @Override

